tests/benchmark_tests/solid_3D_ele/4C_solid_3D_ele_evaluation_benchmark.cpp

```diff
@@ -0,0 +1,203 @@
+#include "4C_global_data.hpp"
+#include "4C_inpar_structure.hpp"
+#include "4C_io_input_parameter_container.hpp"
+#include "4C_io_input_parameter_container.templates.hpp"
+#include "4C_linalg_utils_sparse_algebra_create.hpp"
+#include "4C_mat_material_factory.hpp"
+#include "4C_mat_par_bundle.hpp"
+#include "4C_solid_3D_ele_properties.hpp"
+#include "4C_utils_singleton_owner.hpp"
+
+#include <benchmark/benchmark.h>
+#include <Teuchos_ParameterList.hpp>
+
+#include <array>
+#include <cstddef>
+#include <cstdlib>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+FOUR_C_NAMESPACE_OPEN
+
+namespace
+{
+  void setup_material_in_global_problem()
+  {
+    Core::IO::InputParameterContainer container{};
+    container.add("YOUNG", 1.0);
+    container.add("NUE", 0.1);
+    container.add("DENS", 0.1);
+    container.add("THEXPANS", 0.1);
+
+    // add material to problem instance
+    const int probinst = Global::Problem::instance()->materials()->get_read_from_problem();
+    Global::Problem::instance(probinst)->materials()->insert(
+        1, Mat::make_parameter(1, Core::Materials::MaterialType::m_stvenant, container));
+  }
+
+  template <Core::FE::CellType celltype>
+  void add_nodes_to_discretization(Core::FE::Discretization& dis)
+  {
+    const auto add_node_and_coords = [&](int id, std::vector<double> coords)
+    { dis.add_node(std::make_unique<Core::Nodes::Node>(id, coords, 0)); };
+
+    Core::LinAlg::SerialDenseMatrix reference_nodes =
+        Core::FE::get_ele_node_numbering_nodes_paramspace(celltype);
+    for (int i = 0; i < reference_nodes.num_cols(); ++i)
+    {
+      std::vector<double> coord(3);
+      for (int j = 0; j < Core::FE::dim<celltype>; ++j)
+      {
+        // create some deterministic coordinates
+        coord[j] = reference_nodes(j, i) + 0.01 * (std::sin(541.0 * i) + 0.05 * std::cos(41.0 * j));
+      }
+      add_node_and_coords(i + 1, coord);
+    }
+  }
+
+  template <Core::FE::CellType celltype>
+  std::shared_ptr<Core::Elements::Element> make_solid_element(
+      const Discret::Elements::ElementTechnology& element_technology,
+      Inpar::Solid::KinemType kinem_type)
+  {
+    constexpr auto ele_type = "SOLID";
+    std::array<int, Core::FE::num_nodes(celltype)> node_ids{};
+    for (std::size_t i = 0; i < Core::FE::num_nodes(celltype); ++i) node_ids[i] = i + 1;
+
+    std::shared_ptr<Core::Elements::Element> ele =
+        Core::Communication::factory(ele_type, Core::FE::celltype_string<celltype>, 1, 0);
+    ele->set_node_ids(Core::FE::num_nodes(celltype), node_ids.data());
+
+
+    Core::IO::InputParameterContainer container{};
+    container.add("MAT", 1);
+    container.add("KINEM", kinem_type);
+    container.add("TECH", element_technology);
+
+    ele->read_element(ele_type, Core::FE::cell_type_to_string(celltype), container);
+
+    return ele;
+  }
+
+  template <Core::FE::CellType celltype>
+  std::unique_ptr<Core::FE::Discretization> make_discretization(
```

Comment:
We have such helpers in a few places already and should probably merge them somewhere. The deal.II tests and rebalancing tests come to mind, but there are probably more.